Sources/LiveKit/Publications/RemoteTrackPublication.swift

```diff
@@ -115,11 +117,29 @@ public class RemoteTrackPublication: TrackPublication {
     public func set(preferredFPS newValue: UInt) -> Promise<Void> {
         // no-op if already the desired value
         let trackSettings = _state.trackSettings
-        guard trackSettings.preferredFPS != newValue else { return Promise(()) }
+        guard trackSettings.fps != newValue else { return Promise(()) }
+
+        guard userCanModifyTrackSettings else { return Promise(TrackError.state(message: "adaptiveStream must be disabled and track must be subscribed")) }
+
+        let settings = trackSettings.copyWith(fps: newValue)
+        // attempt to set the new settings
+        return send(trackSettings: settings)
+    }
+
+    /// For tracks that support simulcasting, adjust subscribed quality.
+    ///
+    /// This indicates the highest quality the client can accept. if network
+    /// bandwidth does not allow, server will automatically reduce quality to
+    /// optimize for uninterrupted video.
+    @discardableResult
+    public func set(preferredVideoQuality newValue: VideoQuality) -> Promise<Void> {
```

Comment:
it'd be better to support the dimensions API instead of video quality. we would like to deprecate video quality settings because it's not precise (i.e. if the publisher only publishes two layers, they'd be mapped to LOW/MED). it would not make sense for subscriber to request HIGH